clusterloader2/pkg/test/modules.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+// flattenModuleSteps computes a slice of executable steps. An executable step
+// is either a measurement or phase step. The method recursively flattens the
+// steps being part of a module while skipping the module steps itself. E.g. if
+// the input is:
+// - step 1:
+// - step 2:
+//   module:
+//     - step 3:
+//     - step 4:
+//       module:
+//         -step 5
+// - step 6
+// the method will return [step1, step3, step5, step6].
```

Comment:
I don't understand this:
- where is step2 ?
- where is step4 ?

BTW - it would be good to accomodate the comment to the new API.

Reply:
`step 2` and `step 4` are modules and therefore they're formally steps that are not executable but wrap some other steps (at least that's my understanding here).

Reply:
Yes, Antek explained it well. 

Changed the comment.